src/doc/rustc/src/target-tier-policy.md

```diff
@@ -122,12 +122,22 @@ To propose addition of a new target, open a pull request on [`rust-lang/rust`]:
   r? compiler
   ```
 
+See also the documentation in the `rustc-dev-guide` on [adding a new target to
+`rustc`][rustc_dev_guide_add_target].
+
+Note that crates like `libc` and `cc` may not accept your patches before the
+`rustc` target specification itself exists, so it is recommended that you
+start out with a minimal pull request adding just that (i.e. only
+`#![no_core]` support), and then slowly expand the support in later pull
+requests.
```

Comment:
Hang on, doesn't sometimes r-l/r side PR need PRs to `libc` and `cc` to unblock the r-l/r side too? Won't this be a cyclic requirement then deadlock?

Reply:
Ohh I see, this is explicitly to break *that* cycle right?

Reply:
Exactly. Not happy with how it's written, open to suggestions for better wording.

Reply:
I'll open a T-compiler thread about this to clarify how to handle this cycle when trying to add targets

Reply:
https://rust-lang.zulipchat.com/#narrow/channel/131828-t-compiler/topic/Improving.20docs.20for.20adding.20new.20targets, please feel free to add additional details as you see fit

Reply:
I'll compiler-nominate this for awareness.

Reply:
cc @NobodyXu in case you would like to add anything about adding new targets on the cc-rs side

Reply:
> Exactly. Not happy with how it's written, open to suggestions for better wording.

Maybe something like

> Note that adding a new target that wants to support `std` would transitively require `cc` and `libc` support. However, `cc` and `libc` would need to know about the target from `rustc` as well. To break this cycle, you are strongly encouraged to add a *minimal* `#![no_core]` target spec first to teach `rustc` about the target's existence *first*, which `cc` and `libc` could then learn from, and add `std` support as a follow-up.

Let me know what you think.

Reply:
I've rewritten it to something of that effect.

@rustbot ready